source/drivers/MicroBitFile.cpp

```diff
@@ -0,0 +1,280 @@
+#include "MicroBitFile.h"
+#include "ErrorNo.h"
+
+/**
+  * Creates an instance of a MicroBitFile and creates a new file if required.
+  *
+  * @param fileName the name of the file to create/open.
+  *
+  * @param mode One of: READ, WRITE, READ_AND_WRITE. Defaults to READ_AND_WRITE.
+  */
+MicroBitFile::MicroBitFile(ManagedString fileName, int mode)
+{
+    this->fileName = fileName;
+
+    MicroBitFileSystem* fs;
+
+    if(MicroBitFileSystem::defaultFileSystem == NULL)
+        fs = new MicroBitFileSystem();
+    else
+        fs = MicroBitFileSystem::defaultFileSystem;
+
+    fileHandle = fs->open(fileName.toCharArray(), mode | CREATE);
```

Comment:
Given that we want to avoid exceptions, C++ doesn't give us any nice way of dealing with this.  I'd be happy with an is_valid() or equivalent method that checks the filehandle value and returns a boolean, allowing

```cpp
MicroBitFile f("hello.txt");
if (!f.is_valid()) {
   // handle error
}